src/common/utils.lua

```diff
@@ -333,4 +333,42 @@ function utils.affiliationsForAddress(address, ants)
 	return affiliations
 end
 
+--- @param versionString string
+function utils.validateSemver(versionString)
+	assert(type(versionString) == "string", "Version must be a string")
+
+	-- Reject versions with build metadata (anything after +)
+	if versionString:find("%+") then
+		error("Build metadata is not supported: " .. versionString)
+	end
+
+	-- The pattern now ensures there are exactly 3 version segments
+	-- and optionally a prerelease identifier
+	local pattern = "^(%d+)%.(%d+)%.(%d+)$"
+	local prereleasePattern = "^(%d+)%.(%d+)%.(%d+)%-([%w%.%-]+)$"
```

Comment:
I'm having trouble understanding the purpose of the `%-` at the end of this set 🤔  `[%w%.%-]`. What's the motivation?

Reply:
Will remove - stepping away from semver to iterative versions